### PR TITLE
PP3 regression tests

### PR DESCRIPTION
**Pull request by mkurc-ant**
This PR adds a mechanism for adding design implementation results checking. This includes
 - Block usage
 - Timing analysis (CPD, Fmax)

Along with this PR tests for a counter-like design were added that take advantage of the added checking mechanism. The tests run both for the CMake flow and the installed toolchain.

### CODE REVIEW

File: common/cmake/devices.cmake [hunk 1]
```diff
@@ -2134,7 +2137,7 @@ function(ADD_FPGA_TARGET)
   # Generate routing.
   # -------------------------------------------------------------------------
   add_custom_command(
-    OUTPUT ${OUT_ROUTE}
+    OUTPUT ${OUT_ROUTE} ${OUT_LOCAL}/route.log
```
Comment (lnsharma) [inline common/cmake/devices.cmake:2140]: I see route.log being called multiple times. Should we replace it with a variable?

File: quicklogic/pp3/tests/features/install_test/counter/Makefile.symbiflow [hunk 2]
```diff
@@ -0,0 +1,50 @@
+.PHONY:${BUILDDIR}
+
+current_dir := /home/mkurc/Repos/quicklogic-symbiflow-arch-defs/quicklogic/pp3/tests/features/install_test/counter
```
Comment (lnsharma) [inline quicklogic/pp3/tests/features/install_test/counter/Makefile.symbiflow:3]: This is pointing to local path. In this file there are multiple such instances.

File: quicklogic/pp3/tests/features/install_test/counter_assp/Makefile.symbiflow [hunk 3]
```diff
@@ -0,0 +1,68 @@
+.PHONY:${BUILDDIR}
+
+current_dir := /home/mkurc/Repos/quicklogic-symbiflow-arch-defs/quicklogic/pp3/tests/features/install_test/counter_assp
```
Comment (lnsharma) [inline quicklogic/pp3/tests/features/install_test/counter_assp/Makefile.symbiflow:3]: This is pointing to local path. In this file there are multiple such instances.

File: quicklogic/pp3/tests/features/install_test/counter/Makefile.symbiflow (hunk 2 above)
Comment (mkurc-ant) [reply quicklogic/pp3/tests/features/install_test/counter/Makefile.symbiflow:3]: I have added this file by accident. It shouldn't be here. I'm going to remove it.

File: quicklogic/pp3/tests/features/install_test/counter_assp/Makefile.symbiflow (hunk 3 above)
Comment (mkurc-ant) [reply quicklogic/pp3/tests/features/install_test/counter_assp/Makefile.symbiflow:3]: I have added this file by accident. It shouldn't be here. I'm going to remove it.

File: common/cmake/devices.cmake (hunk 1 above)
Comment (mkurc-ant) [reply common/cmake/devices.cmake]: Right, I added a variable.

**Review by lnsharma — APPROVED** [review]
LGTM

**Pull request merged**